src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -36,13 +38,56 @@
      *
      * @return Reference[]
      */
-    private function findAndSortTaggedServices($tagName, ContainerBuilder $container)
+    private function findAndSortTaggedServices($tagName, ContainerBuilder $container, string $indexAttribute = null, string $defaultIndexMethod = null)
     {
         $services = array();
 
+        if (null === $indexAttribute && null !== $defaultIndexMethod) {
```

Comment:
is this really required? i might want to "key by method" as a default approach

Reply:
It is: one must have a way to configure the key they want using configuration. The command example is again enlightening: it *must* be possible to ignore the default command name and let configuration take over the real command name. Same here.